pkg/logging/logger.go

```diff
@@ -0,0 +1,46 @@
+package logging
+
+import (
+	"os"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+const (
+	LogFormatLogfmt = "logfmt"
+	LogFormatJSON   = "json"
+)
+
+func NewLogger(logLevel, logFormat, debugName string) log.Logger {
+	var (
+		logger log.Logger
+		lvl    level.Option
+	)
+
+	switch logLevel {
+	case "error":
+		lvl = level.AllowError()
+	case "warn":
+		lvl = level.AllowWarn()
+	case "info":
+		lvl = level.AllowInfo()
+	case "debug":
+		lvl = level.AllowDebug()
+	default:
+		panic("unexpected log level")
```

Comment:
I realize that we had the same behavior before, but I don't think that panicking is the right action to take when a user mistypes a flag. If this was only possible in a program error, then I would agree, but since it's possible when typing `--log.level=eror` I think it's too severe.

Reply:
Refactored this a bit, let me know what you think about this approach.